File: Day<17>/Solution.cpp
#include <bits/stdc++.h> 
using namespace std; 


 // } Driver Code Ends
//User function Template for C++

class Solution{
    
    public:
    vector<int> help_classmate(vector<int> arr, int n) 
    { 
    	stack<int> s; 
    	vector<int> result(n,-1);
    	
    	for (int i = 0; i < n; i++) { 
    		if (s.empty()) { 
    			s.push(i); 
    			continue; 
    		} 
    		while (s.empty() == false && arr[s.top()] > arr[i]) { 
    			result[s.top()] = arr[i];
    			s.pop(); 
    		} 
    		s.push(i); 
    	} 
    	return result;
    } 
};

// { Driver Code Starts.

int main() 
{ 
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		vector<int> array(n);
		for (int i = 0; i < n; ++i)
		{
			cin>>array[i];
		}
		Solution obj;
		vector<int> result = obj.help_classmate(array,n);
		for (int i = 0; i < n; ++i)
		{
			cout<<result[i]<<" ";
		}
		cout<<"\n";
	}
	return 0; 
}

File: Day<17>/Solution.java
import java.util.*; 
import java.io.*; 

class GFG 
{ 
	public static void main (String[] args) { 
		Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t > 0){
        	int n = sc.nextInt();
            int array[] = new int[n];
            for (int i = 0; i < n; ++i)
            {
                array[i] = sc.nextInt();
            }

            Solution ob = new Solution();

            int ans[] = ob.help_classmate(array,n);

           	for (int i=0; i<n; i++) 
                System.out.print(ans[i]+" "); 
            System.out.println();
            t--;
        }
	} 
} 

// } Driver Code Ends

class Solution {
	public static int[] help_classmate(int arr[], int n) 
	{ 
		Stack<Integer> s = new Stack<Integer>(); 
		int result[] = new int[n];
 		Arrays.fill(result, -1); 

		for (int i = 0; i < n; i++) { 

			if (s.empty()) { 
				s.push(i); 
				continue; 
			} 

			while (s.empty() == false && arr[s.peek()] > arr[i]) { 
				result[s.peek()] = arr[i]; 
				s.pop(); 
			} 
			s.push(i); 
		} 
		return result; 
	} 	
}
